pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -921,6 +923,16 @@ cdef class MultiIndexHashTable(HashTable):
                     "hash collision\nlocs:\n{}\n"
                     "result:\n{}\nmi:\n{}".format(alocs, result, mi))
 
+    def _check_for_collision(self, Py_ssize_t loc, object label):
+        # validate that the loc maps to the actual value
+        # version of _check_for_collisions above for single label (tuple)
+
+        result = self.mi[loc]
+        if not array_equivalent(result, label):
```

Comment:
Actually, I wanted to test whether this comparison was working on a real usecase with NaNs in the index, but this seems to *not* work anyway (irregardless of this collision check):

```
In [11]: mi = pd.MultiIndex.from_product([['A', 'B'], [1, np.nan]])

In [12]: mi
Out[12]: 
MultiIndex(levels=[['A', 'B'], [1]],
           labels=[[0, 0, 1, 1], [0, -1, 0, -1]])

In [13]: mi.get_loc(('A', np.nan))
...
pandas/src/hashtable_class_helper.pxi in pandas.hashtable.PyObjectHashTable.get_item (pandas/hashtable.c:13696)()

KeyError: ('A', nan)
```

The above is with 0.19.2, so this was already failing with the PyObject hashtable